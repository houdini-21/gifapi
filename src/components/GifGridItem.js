import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({ title, url }) => {
  return (
    <div className="card__item  animate__animated animate__fadeInUp">
      <img src={url} alt={title} className="card__item-img" />
      <p className="card__item-title">{title}</p>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default GifGridItem;
