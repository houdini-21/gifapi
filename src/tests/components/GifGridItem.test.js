import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('pruebas en gridgif', () => {
  let wrapper = shallow(
    <GifGridItem title="silicon valley" url="https://github.com/houdini-21" />
  );
  test('debe de mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de tener un parrafo en <p>', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe('silicon valley');
  });

  test('debe de tener un img igual a https://github.com/houdini-21', () => {
    const url = wrapper.find('img');
    expect(url.props().src).toBe('https://github.com/houdini-21');
    expect(url.props().alt).toBe('silicon valley');
  });

  test('debe de contener la clase animate__fadeInUp', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect(className.includes('animate__fadeInUp')).toBe(true);
  });
});
