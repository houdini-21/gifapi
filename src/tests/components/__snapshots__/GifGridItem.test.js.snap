// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en gridgif debe de mostrar el componente correctamente 1`] = `
<div
  className="card__item  animate__animated animate__fadeInUp"
>
  <img
    alt="silicon valley"
    className="card__item-img"
    src="https://github.com/houdini-21"
  />
  <p
    className="card__item-title"
  >
    silicon valley
  </p>
</div>
`;
